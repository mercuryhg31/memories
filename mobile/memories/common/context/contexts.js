import { createContext, useContext, useState } from 'react';

const CurrentUserContext = createContext();
function CurrentUserContextProvider({ children }) {
  const [currentUserID, setCurrentUser] = useState('');
  const [displayUser, setDisplayUser] = useState(false);
  const [targetUserUID, setTargetUserUID] = useState('');
  const endpointURL = 'https://reqres.in';

  return (
    <CurrentUserContext.Provider
      value={{
        currentUserID,
        setCurrentUser,
        displayUser, // boolean of if the profile modal should be displayed
        setDisplayUser,
        targetUserUID,
        setTargetUserUID,
        endpointURL,
      }}>
      {children}
    </CurrentUserContext.Provider>
  );
};

export { CurrentUserContext, CurrentUserContextProvider };