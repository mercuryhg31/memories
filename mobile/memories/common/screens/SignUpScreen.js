import { useState, useEffect, createContext, useContext, } from 'react';
import { StyleSheet, View, Text, SafeAreaView, Alert, Image, TouchableOpacity, Dimensions } from 'react-native';
import CustomInput from '../components/customInput.component';
import CustomButton from '../components/customButton.component';
import { CurrentUserContext } from '../context/contexts';
import {launchCamera, launchImageLibrary} from 'react-native-image-picker';
import { createUserSuccessful } from '../helpers/requestHelpers';

import * as ImagePicker from 'expo-image-picker';

const SignUpScreen = ({ navigation }) => {
    const { setCurrentUser } = useContext(CurrentUserContext);

    // handle log in functionality and only pass up the user name
    [userName, setUserName] = useState('');
    [password, SetPassword] = useState('');
    [email, SetEmail] = useState('');
    [bio, setBio] = useState('');

    const attemptSignUp = () => {

        if(userName === ''){
            Alert.alert('Enter userName', 'login requires userName', [
                {text: 'OK'},                
              ]);
        } else if(email === ''){
            Alert.alert('Enter email', 'login requires email', [
                {text: 'OK'},                
              ]);
        } else if(password === ''){
            Alert.alert('Enter password', 'login requires password', [
                {text: 'OK'},                
              ]);
        } else{

            // create and login user if possible
            createUserSuccessful(userName, email, password, bio).then((userLoginStatus) => {
                if(userLoginStatus){ // valid login

                    // should parse request for userid to make future requests
                    setCurrentUser(email); // save email for future requests (temporary solution)
                    navigation.navigate('MainScreen'); // navigate to map
                }else{
                    console.warn('no account exists with that email & password.');
                }
            }).catch(
                (err) => {console.log(err)});

            if (true){ // valid login (replace with request)
                navigation.navigate('MainScreen'); // navigate to map
                setCurrentUser(userName); // save username for future requests
            } else{
                console.warn('cannot create an account with that information.')
                // TODO: parse to provide informative error of why account creation didn't succeed
            }
        }
    }

    const [userIcon, setUserIcon] = useState('https://cdn-icons-png.flaticon.com/512/3177/3177440.png');
    
    // const ImagePicker = async () => {
    //     let options = {
    //         storageOptions: {
    //             path: "image"
    //         }
    //     }

    //     launchImageLibrary(options, (response)=>{
    //         setUserIcon(response.assets[0].uri);
    //     }).then(console.log('it was a dark and stormy night')).catch(alert('u suck'));
    // }

    const pickImageAsync = async () => {
        let result = await ImagePicker.launchImageLibraryAsync({
            allowsEditing: true,
            quality: 1
        });

        if (!result.canceled) {
            setUserIcon(result.assets[0].uri);
        } else {
            alert("pick an image dude");
        }
    };

    return <View style={styles.container}>

        <SafeAreaView >
            <View /* Logo container */ style={styles.logo_container}>
                <TouchableOpacity onPress={pickImageAsync}>
                    <Image
                        style={styles.logo}
                        source={{
                            uri: userIcon,
                        }}
                    />
                </TouchableOpacity>
            </View>

            <View /* Registration container */ style={styles.inputContainer}>
                <CustomInput
                    placeholder={'username'}
                    setValue={setUserName}
                    />
                <CustomInput
                    placeholder={'email'}
                    setValue={SetEmail}
                    />
                <CustomInput
                    placeholder={'password'}
                    setValue={SetPassword}
                    />
                <CustomInput
                    placeholder={'[optional]Bio  '}
                    setValue={SetPassword}
                    isMuliLine={true}
                    />
                <CustomButton
                    placeholder={'Create Account'}
                    onPress={() => attemptSignUp()}
                    button_type={styles.createAccountBtn}
                    />

            </View>

        </SafeAreaView>
    </View>
};

const vh = Dimensions.get('window').height;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#c5effc',
    },
    safe_container: {
        flex: 1,
        flexDirection: 'column',
    },
    logo_container: {
        alignItems: "center",
        paddingTop: .05*vh,
    },
    logo: {
        width: .25*vh, height: .25*vh,
        borderRadius: 100
    },
    inputContainer: {
        paddingTop: .05*vh,
        alignItems: "center",
        gap: 0.025*vh,
    },

    createAccountBtn: {
        backgroundColor: '#45d9a8',
        width: '80%',
        borderColor: '#57ab8f',
        borderWidth: 1,
        borderRadius: 5,
        paddingVertical: 10,
        alignItems: "center",
    },
});

export default SignUpScreen;