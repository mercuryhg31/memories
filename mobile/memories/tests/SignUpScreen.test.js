import React from 'react';
import renderer from 'react-test-renderer';
import { CurrentUserContextProvider } from '../common/context/contexts';
import SignUpScreen from '../common/screens/SignUpScreen';


describe('<SignUpScreen/>', () => {
  test('has 1 child', () => {
    const tree = renderer.create(<CurrentUserContextProvider><SignUpScreen/></CurrentUserContextProvider>).toJSON();
    expect(tree.children.length).toBe(1);
  });

  test('renders correctly', () => {
    const tree = renderer.create(<CurrentUserContextProvider><SignUpScreen/></CurrentUserContextProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});