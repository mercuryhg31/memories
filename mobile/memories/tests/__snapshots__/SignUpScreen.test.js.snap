// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpScreen/> renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#c5effc",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView>
    <View
      style={
        {
          "alignItems": "center",
          "paddingTop": 66.7,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://cdn-icons-png.flaticon.com/512/3177/3177440.png",
            }
          }
          style={
            {
              "borderRadius": 100,
              "height": 333.5,
              "width": 333.5,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 33.35,
          "paddingTop": 66.7,
        }
      }
    >
      <View
        accessibilityLabel="Username input field"
        onLayout={[Function]}
        style={
          [
            {
              "backgroundColor": "white",
              "borderColor": "#e8e8e8",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
              "width": "80%",
            },
            {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="username"
          style={{}}
        />
      </View>
      <View
        accessibilityLabel="Email input field"
        onLayout={[Function]}
        style={
          [
            {
              "backgroundColor": "white",
              "borderColor": "#e8e8e8",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
              "width": "80%",
            },
            {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="email"
          style={{}}
        />
      </View>
      <View
        accessibilityLabel="Password input field"
        onLayout={[Function]}
        style={
          [
            {
              "backgroundColor": "white",
              "borderColor": "#e8e8e8",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 40,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
              "width": "80%",
            },
            {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="password"
          style={{}}
        />
      </View>
      <View
        accessibilityLabel="Bio input field"
        onLayout={[Function]}
        style={
          [
            {
              "backgroundColor": "white",
              "borderColor": "#e8e8e8",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 110,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
              "width": "80%",
            },
            {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="[optional]Bio  "
          style={{}}
        />
      </View>
      <View
        accessibilityLabel="Create account button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#45d9a8",
            "borderColor": "#57ab8f",
            "borderRadius": 5,
            "borderWidth": 1,
            "opacity": 1,
            "paddingVertical": 10,
            "width": "80%",
          }
        }
      >
        <Text>
          Create Account
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
